services/event/service/event_service.go

```diff
@@ -487,42 +487,44 @@ func CanRedeemPoints(event_code string) (bool, string, error) {
 	err := db.FindOne("eventcodes", query, &eventCode)
 
 	if err != nil {
-		return false, "invalid", err
+		return false, false, "invalid", err
 	}
 
 	expiration_time := eventCode.Expiration
 	current_time := time.Now().Unix()
 
-	return current_time < expiration_time, eventCode.ID, nil
+	return current_time < expiration_time, eventCode.IsVirtual, eventCode.ID, nil
 }
 
 /*
 	Returns the eventcode struct for the event with the given id
 */
-func GetEventCode(id string) (*models.EventCode, error) {
+func GetEventCode(id string) (*[]models.EventCode, error) {
```

Comment:
we should change the name of this function to `GetEventCodes` since it returns a list now